enhancements/ingress/customize-error-code-pages.md

```diff
@@ -0,0 +1,146 @@
+It's also important that the right page as per user requirements is served at the right moment.
+Example - 503 error page shall be returned if no pod is available. 
+But when the URI is wrong/does not exist, we should return 404 or a customized 404 page as per user requirements.
+Using this feature, user can provide custom http pages for 503 and 404 for the haproxy router by adding these pages
+in a configmap using --file attribute.
+## Motivation
+Users say 503 page is served even when the page is not found when it shall serve 404. 
+The reason that happens is because we provide only 503 in our haproxy template.
+
+### Goals
+1. Enable the cluster administrator to specify custom error code pages for haproxy router to serve them at the right moment.
+
+### Non-Goal
+TBD
+
+## Proposal
+0. User creates a custom error code page configmap `my-custom-error-code-pages` in `openshift-config`
+1. The IngressController API is extended by adding an optional
+`HttpErrorCodePages` field with type string
+```go
+	// httpErrorCodePages specifies a configmap with custom error pages.
+	// The administrator must create this configmap in the openshift-config namespace.
+	// This configmap should have keys in the format "error-page-<error code>.http",
+	// where <error code> is an HTTP error code.
+	// For example, "error-page-503.http" defines an error page for HTTP 503 responses.
+	// Currently only error pages for 503 and 404 responses can be customized.
+	// Each value in the configmap should be the full response, including HTTP headers.
+	// If this field is empty, the ingress controller uses the default error pages.
+	HttpErrorCodePages string `json:"httpErrorCodePages,omitempty"`
+```
+Using the default, the router uses the default error pages used by haproxy.
+The `HttpErrorCodePage` field enables the cluster administrator to
+specify custom error code pages.
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  httpErrorCodePage: "my-custom-error-code-pages"  // my-custom-error-code-pages is the config map name with custom 
+                                                   //http error pages, created in openshift-config by cluster-admin. 
+```
+2. The configmap keys shall have the format error-page-`<error code>`.http. Right now only error-page-503.http and 
+   error-page-404.http will be available.
+```yaml
+$ oc -n openshift-config create configmap  my-custom-error-code-pages\
+--from-file=error-page-503.http\
+--from-file=error-page-404.http
+
+   $ oc get configmaps my-custom-error-code-pages -o yaml
+
+apiVersion: v1
+data:
+   error-page-503.http: |-
+     <your custom http response page contents>
+   error-page-404.http: |
+     <your custom http response page contents>
+kind: ConfigMap
+
+```  
+3. `errorfile` stanzas need to be added as per https://www.haproxy.com/blog/serve-dynamic-custom-error-pages-with-haproxy/
```

Comment:
```suggestion
3. In response to the user configuration, the IngressController will configure `errorfile` stanzas as needed in the IngressController's `haproxy.config` file per https://www.haproxy.com/blog/serve-dynamic-custom-error-pages-with-haproxy/
```
This is a little confusing because it seems like you doing two things at once:  You are both describing the procedure that the user would follow to use this feature, and you are also outlining the steps that the developer (you) will take to implement the feature.